packages/extension-api/src/extension-api.d.ts

```diff
@@ -788,6 +789,93 @@ declare module '@podman-desktop/api' {
     connection: ContainerProviderConnection;
   }
 
+  /**
+   * Callback for openning shell session
+   */
+  export interface ProviderConnectionShellAccess {
+    /**
+     * Opens new session using ProviderConnectionShellAccessImpl class
+     * @example
+     * ```typescript
+     * const providerConnectionShellAccess = new ProviderConnectionShellAccessImpl(machineInfo);
+     * const session = providerConnectionShellAccess.open()
+     * ```
+     */
+    open(): ProviderConnectionShellAccessSession;
```

Comment:
not sure I get the example.

If I look at the example part, it's the code I'll write in my extension so my extension will never call the open method, it will be Podman Desktop